src/mongo_client.ts

```diff
@@ -380,6 +404,15 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
       sessionPool: new ServerSessionPool(this),
       activeSessions: new Set(),
 
+      getAuthProvider(name: AuthMechanism | string) {
```

Comment:
I find it a little bit unexpected that this method resides on the client's internal state object. Could we possibly just make it a floating function in its own module and mark it @internal and then move the `AUTH_PROVIDERS` map into that module as well?

Reply:
Considering that the method looks weird in the client's internal state object, I refactored it a bit and moved the auth providers cache to the separate MongoClientAuthProviders module. We can't completely get rid of it as a part of the MongoClient state, because we need to create providers per client instead of per module. We could make getAuthProvider a floating function, but then we need to pass the client reference to it from the connection pool and monitoring instances to get access to existing client providers, and then pass it all along across different helper functions like performInitialHandshake() and prepareHandshakeDocument(), which all have access to ConnectionOptions. Another thing to consider is that we can't decide on a single provider when we create a MongoClient instance because a client can have more than one auth provider, - while connecting we can e.g. switch from MONGODB_DEFAULT to MONGODB_SCRAM_SHA256. Let me know if this sounds reasonable or if you have thoughts on how to refactor it further.